src/app/Plugins/Weather/Components/weather/weather.component.ts

```diff
@@ -26,6 +29,20 @@ export class WeatherComponent implements OnInit {
     this.today = false;
     this.tommorow = false;
     this.week = false;
+    this.weatherService.getAllLocations()
+      .subscribe((data: Location[]) => {
+        this.locations = data;
+        this.selectedLocation = this.locations[0].name;
+      });
+  }
+
+  public onChange(event): void {
+    console.log(event.value);
+    this.weatherService.sendLocation(event.value);
+
+    if (this.today) { this.getWeatherToday(); }
+    if (this.tommorow) { this.getWeatherTomorrow(); }
+    if (this.week) { this.getWeatherWeek(); }
```

Comment:
This boolean logic is very strange. Doesn't look right.

Reply:
this checks if week is not null, not undefined, not 0 or not false